tests/ccl_test_redshifts.py

```diff
@@ -0,0 +1,149 @@
+import numpy.testing
+import pyccl as ccl
+
+# Set tolerances
+TOLERANCE = 1e-4
+
+def check_redshift_analytic():
+    """
+    Compare the redshift functions to an analytic toy case.
+    """
+    # Redshift input
+    z_lst = [0., 0.5, 1., 1.5, 2.]
+
+    # p(z) function for which we can calculate and analytic dNdz_tomog
+    # when pairs with the toy dndz_ana below
+    def pz_ana(z_ph, z_s, args):
+        return (np.exp(- (z_ph - z_s)**2. / 2. / 0.1**2) / np.sqrt(2.
+                * np.pi) / 0.1)
+
+    # PhotoZFunction class
+    PZ_ana = ccl.PhotoZFunction(pz_ana)
+        
+    # Introduce an unrealistic, simple true function for dNdz for which
+    # we can calculate dNdz_tomog analytically for comparison.
+    def dndz_ana(z, args):
+        if ((z>=0.) and (z<=1.0)): 
+            return 1.0
+        else:
+            return 0. 
+	
+    # import math erf and erfc functions:
+    from math import erf, erfc
+    # And introduce a function to return the analytic result of using 
+    # dndz_ana with a Gaussian p(z,z'), to which we compare
+    def dNdz_tomog_analytic(z, sigz, zmin, zmax):
+        if ( (z>=0.) and (z<=1.0) ):
+            return (erf((z-zmin) / np.sqrt(2.)/sigz) - erf((z-zmax)/
+                    np.sqrt(2.)/sigz)) / (-1. + (-np.exp(-(zmax-1)**2 
+                    / 2. / sigz**2) + np.exp(-zmax**2 / 2. / sigz**2) + 
+                    np.exp(-(zmin-1)**2 / 2. / sigz**2) - np.exp(
+                    -zmin**2 / 2 /sigz**2)) * np.sqrt(2. / np.pi)*sigz 
+                    + erf( (zmax-1) / np.sqrt(2.) /  sigz) + erfc( 
+                    (zmin-1.)/np.sqrt(2.)/sigz) + zmax * (erf(zmax/
+                    np.sqrt(2.)/sigz) - erf( (zmax-1.) / np.sqrt(2.)/
+                    sigz)) + zmin * ( erf( (zmin-1.)/np.sqrt(2.)
+		            / sigz) - erf(zmin/np.sqrt(2.)/sigz)))
+        else:
+            return 0.
+        
+    # dNdzFunction class
+    dNdZ_ana = ccl.dNdzFunction(dndz_ana)
+    
+    # Check that for the analytic case introduced above, we get the 
+    # correct value.
+    zmin = 0.
+    zmax = 1.
+    # math erf funcs are not vectorized so loop over z values
+    for z in z_lst:
+		assert_allclose(ccl.dNdz_tomog(z, zmin, zmax, PZ_ana, 
+		               dNdZ_ana), dNdz_tomog_analytic(z, 0.1, zmin, 
+		               zmax), rtol=TOLERANCE)       
+		            
+def check_redshift_numerical():
+    """
+    Compare the redshift functions to a high precision integral.
+    """
+    # Redshift input
+    z_lst = [0., 0.5, 1., 1.5, 2.]
+                
+     # p(z) function for dNdz_tomog
+    def pz1(z_ph, z_s, args):
+        return np.exp(- (z_ph - z_s)**2. / 2.)
+
+    # Lambda function p(z) function for dNdz_tomog
+    pz2 = lambda z_ph, z_s, args: np.exp(-(z_ph - z_s)**2. / 2.)
+    
+    # Set up a function equivalent to the PhotoZGaussian
+    def pz3(z_ph, z_s, sigz):
+		sig = sigz*(1.+ z_s)
+		return (np.exp(- (z_ph - z_s)**2. / 2. / sig**2) / np.sqrt(2.
+		        *np.pi) / sig)
```

Comment:
A couple places (like here) you have 8 space indents.  It's not wrong, but probably worth keeping to the PEP8 style of 4 spaces.

Reply:
Good point, I've tried to clean this up.